features/cellular/framework/targets/RiotMicro/AT/RM1000_AT_CellularNetwork.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2018, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdlib.h>
+#include "CellularCommon.h"
+#include "RM1000_AT_CellularNetwork.h"
+#include "platform/mbed_wait_api.h"
+
+#include "mbed-trace/mbed_trace.h"
+#ifndef TRACE_GROUP
+#define TRACE_GROUP  "RIOT"
+#endif // TRACE_GROUP
+using namespace mbed;
+
+// Callback for MODEM faults URC.
+void RM1000_AT_CellularNetwork::MODEM_FAULT_URC()
+{
+    tr_debug("RM1000_AT_CellularNetwork::ASSERTED_URC");
+    _connect_status = NSAPI_STATUS_DISCONNECTED;
+    if (_connection_status_cb) {
+        cell_callback_data_t data;
+        data.error = NSAPI_ERROR_CONNECTION_LOST;
+        _connection_status_cb((nsapi_event_t)CellularDeviceTimeout, (intptr_t)&data);
```

Comment:
Should this be `NSAPI_EVENT_CONNECTION_STATUS_CHANGE` with `NSAPI_STATUS_DISCONNECTED`?

Reply:
done